### DISCUSSION TITLE
raven mobile - socket.io

### DISCUSSION THREAD

**Discussion by Alvish0407** [The-Commit-Company/raven#896] [Q&A] [answered]
does raven mobile support socket.io ?

**Comment by Alvish0407** [comment]
In PWA, while sending messages, I have to explicitly refresh the whole page to see sent message. But that's not the case on desktop though. I think I am missing something.

**Comment by nikkothari22** [comment]
Are you on a local development environment?

**Reply by Alvish0407** [reply]
Yes

**Reply by nikkothari22** [reply]
You would have to check the browser console and see if the socket event is connecting.

**Reply by Alvish0407** [reply]
Socket works fine in desktop /raven_mobile. I'm facing issue on mobile app.

**Reply by Alvish0407** [reply]
I tried inspecting raven mobile on iPhone safari. But I didn't get option to inspect it maybe because it runs on localhost.

**Reply by nikkothari22** [reply]
Oh yeah, it doesn't work well if you're on your phone. We usually develop on our desktop. 

We'll be moving away from the mobile app PWA anyway and only focusing on the web app (and making it responsive)

**Reply by Alvish0407** [reply]
You are saying that raven_mobile(which shares same UI between desktop and mobile) contains different implementations internally ?

**Reply by Alvish0407** [reply]
> Socket works fine in desktop /raven_mobile. I'm facing issue on mobile app.

Because, as i said raven_mobile on desktop works perfectly.

**Reply by nikkothari22** [reply, marked as answer]
No I mean socket connection over the network on iPhone in the local development environment does not work reliably.

**Reply by Alvish0407** [reply]
Will it work on production live server ?

**Reply by nikkothari22** [reply]
Yep.

We're going to move away from the mobile app PWA though and build something native eventually. So we won't be updating it with new features and just work on the web app (/raven) now.
